[18.0][MIG] sale_automatic_workflow_job: Migration to 18.0

chaule97 opened the pull request:
- Because [this commit](https://github.com/OCA/sale-workflow/pull/3105/commits/1d0e79e5accf69e71eef92284726b0e10af5e518) separated the module `stock` from `sale_automatic_workflow`.  I removed everything related to `stock`.

Commit:
[FIX] Add sale order dolder

Commit:
[FIX] adds _run_move_create and some other fixes

Commit:
[FIX] add restriction on sale confirm changed icon and minor fix

[FIX] Fix product_id on test

[FIX] Fix product on test

Commit:
[FIX] Add check on sale order

Commit:
[FIX] add _run override

Commit:
[FIX] adding feature and fix contributors

Commit:
Add transmition of lot_id from move to procurement and delete lot_id on
copy

Commit:
Extract move state check for easier inherit and add copy=False instead
of copy method

Commit:
Fix pep8 and edit check move state

Commit:
Fix unused import

eLBati committed:
[IMP] better lot names in tests

Commit:
[FIX] Fixed Tests adding a stock_picking_in to create quants

[FIX] Use product_14 instead of product_11

[FIX] Add test's flow step

Commit:
[FIX] Add check on lot reservation in test

Commit:
[FIX] Add check on quants

[FIX] change data in quant search

[FIX] Fixed tab

[FIX] Add check on lots list

FIX product_id_change_with_wh

eLBati committed:
Refactoring product_id_change_with_wh for domain generation.
Because quants_get_prefered_domain does not return quants with our lots, if other quants are available

eLBati committed:
Refactoring tests, adding onchange tests

eLBati committed:
Remove never used domain: on_change will set the correct domain for lot_id field

eLBati committed:
IMP module version and fix comment

eLBati committed:
the field lot_id should only be visible to the members of the stock.group_production_lot res.groups

eLBati committed:
REF some code and ADD lot11 has 1 availability and order4 has quantity 2

eLBati committed:
REF product_id_change_with_wh for compatibility with ohter override

eLBati committed:
FIX default value in context

eLBati committed:
IMP README

eLBati committed:
FIX You cannot transfer a picking in state 'False'

oca-transbot committed:
OCA Transbot updated translations from Transifex

sebastienbeau committed:
[FIX] fix perf issue with sale_order_lot_selection

eLBati committed:
FIX https://github.com/OCA/sale-workflow/issues/269

Steps to reproduce:

1.- Create a new Sale Order with 2 or more lines of products
2.- Assign lot to every line (ensure lot has stock available)
3.- Confirm the Sale Order
4.- Go to the Delivery Order

Result:

Only the first line in the Delivery Order is in state "Available" and reserved, the rest of the lines are in state "Waiting Availability" and not reserved

florian-dacosta committed:
Migrate sale_order_lot_selection to v9

florian-dacosta committed:
[FIX] return super result after sale order confirm

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

mourad-ehm committed:
[MIG] sale_order_lot_selection: Migrated to 10.0

lmignon committed:
[FIX] sale_order_lot_selection: _prepare_procurement_from_move is an api.multi method! (#475)

tafaRU committed:
[8.0][ADD] sale_delivery_split_date module (#620)

* [ADD] sale_delivery_split_date module

* [ADD] sale_delivery_split_date: add computed min_dt to allow deliveries to be searchable by the scheduled date

tafaRU committed:
[MIG] sale_delivery_split_date: Migration to 10.0

tafaRU committed:
[FIX] sale_delivery_split_date: fix incompatible modules urls in README file

oca-travis committed:
[UPD] Update sale_order_lot_selection.pot

rousseldenis committed:
[10.0][FIX] pylint

PCatinean committed:
[MIG] sale_order_lot_selection: Migration to 12.0

oca-travis committed:
[UPD] Update sale_order_lot_selection.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_order_lot_selection
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_order_lot_selection/

MiquelRForgeFlow committed:
[ADD] sale_order_warn_message

Commit:
[IMP] : black, isort, prettier

Commit:
[13.0][MIG] sale_order_lot_selection

Commit:
[ADD] maintainers

Commit:
[IMP] add context 'skip_check_lot_selection_qty' to skip check lot selection process

Commit:
[IMP] split files as per each object

Commit:
Update sale_order_lot_selection/readme/DESCRIPTION.rst

Co-Authored-By: Daniel Reis <dreis@opensourceintegrators.com>

Commit:
Add sale_automatic_workflow_job

Use Queue Jobs to process the Sales Automatic Workflow actions.

The default behavior of the automatic workflow module is to use a
scheduled action that searches all the record that need a workflow
action and sequentially process all of them.

It can hit some limits when the number of records is too high.

This module keeps the scheduled action to search the records, but
instead of directly executing the actions (confirm a sales order,
create invoices for a sales order, validate invoices, ...), it
creates one job per operation to do.

It uses an identity key on the jobs so it will not create the same
job for the same record and same operation twice.

~

I needed to extract methods in `sale_automatic_workflow` in order
to be able to execute them as jobs.

A new decorator "job_auto_delay" (to eventually move in queue_job) makes
these methods automatically delayed when called.

oca-travis committed:
[UPD] Update sale_automatic_workflow_job.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

SimoRubi committed:
[MIG] sale_delivery_split_date: Migration to 12.0
Note that sale_delivery_split_date is not compatible with sale_procurement_group_by_commitment_date

Saran440 committed:
[IMP] related_job to origin document

oca-travis committed:
[UPD] Update sale_automatic_workflow_job.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
sale_automatic_workflow_job 12.0.1.0.1

oca-travis committed:
[UPD] Update sale_automatic_workflow_job.pot

kittiu committed:
[IMP] : black, isort, prettier

kittiu committed:
[13.0][MIG] sale_automatic_workflow_job

oca-travis committed:
[UPD] Update sale_order_warn_message.pot

OCA-git-bot committed:
[UPD] README.rst

sbidoul committed:
[FIX] website URL

florian-dacosta committed:
Migrate sale_order_lot_selection to 14
Also add fix from v12 31c9698ec41132996d660d641618591efafeeaef that was not ported to v13

florian-dacosta committed:
[REF] Refactore module making it depend on stock_restrict_lot
The module was buggy, it would reserve stock if available, even if it had the wrong lot. Test added to show the case

florian-dacosta committed:
Remove logic where sale order confirmation is blocked if selected lot is not in stock yet

florian-dacosta committed:
add roadmap

florian-dacosta committed:
[IMP] pre-commit run

oca-travis committed:
[UPD] Update sale_order_lot_selection.pot

OCA-git-bot committed:
[UPD] README.rst

danimv5 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (6 of 6 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_order_lot_selection
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_order_lot_selection/ca/

petrus-v committed:
[14.0] add module: sale_order_note_template

petrus-v committed:
14.0 sale_order_note_template: add post install/uninstall hooks to convert field properly

Rad0van committed:
fix displaying warning, add test for partner with no warning

danimv5 committed:
Added translation using Weblate (Catalan)

danimv5 committed:
Added translation using Weblate (Spanish)

educasilva committed:
Added translation using Weblate (Portuguese (Portugal))

oca-travis committed:
[UPD] Update sale_order_note_template.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
sale_order_warn_message 13.0.1.1.0

mmequignon committed:
Add sale_order_disable_user_autosubscribe

OCA-git-bot committed:
[ADD] icon.png

marylla committed:
Translated using Weblate (German)

Currently translated at 83.3% (5 of 6 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_order_lot_selection
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_order_lot_selection/de/

fredzamoabg committed:
[IMP] sale_delivery_split_date: black, isort, prettier

Commit:
[MIG] sale_order_disable_user_autosubscribe: Migration to 14.0

nicomacr committed:
[14.0][FIX] sale_order_lot_selection: Avoid missing warning in the product selection.

This call of super without a return is cause that alert of warning for product wasn't working.
After this change we ensure that return of warning or domain will apply finally

OCA-git-bot committed:
sale_order_lot_selection 14.0.1.0.1

manuelregidor committed:
[IMP] sale_order_warn_message: black, isort, prettier

manuelregidor committed:
[MIG]sale_order_warn_message: Migration to 14.0

oca-travis committed:
[UPD] Update sale_order_warn_message.pot

OCA-git-bot committed:
[UPD] README.rst

kv1612 committed:
[14.0][MIG] sale_automatic_workflow_job: Migration to 14.0

oca-travis committed:
[UPD] Update sale_automatic_workflow_job.pot

fredzamoabg committed:
[MIG] sale_delivery_split_date: Migration to 14.0

OCA-git-bot committed:
[UPD] README.rst

bealdav committed:
FIX sale__lot_selection: add company ctx

OCA-git-bot committed:
sale_order_lot_selection 14.0.1.1.0

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 33.3% (2 of 6 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_order_lot_selection
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_order_lot_selection/it/

[UPD] README.rst

[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

vimalpatelserpentcs committed:
[ADD] Initial commit

vimalpatelserpentcs committed:
[MIG] sale_automatic_workflow_job: Migration into v15

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 50.0% (3 of 6 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_order_lot_selection
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_order_lot_selection/it/

Commit:
[UPD] Update sale_automatic_workflow_job.pot

OCA-git-bot committed:
[UPD] README.rst

nicomacr committed:
[MIG] sale_order_lot_selection: Migration to 15.0

florian-dacosta committed:
[MIG][16.0] Migrate sale_order_lot_selection to v16

OCA-git-bot committed:
sale_order_lot_selection 16.0.1.0.1

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_delivery_split_date
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_delivery_split_date/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_order_lot_selection
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_order_lot_selection/it/

amarcosg committed:
[MIG] sale_automatic_workflow_job: Migration to 16.0

yankinmax committed:
[MIG] sale_delivery_split_date: Migration to 15.0

Commit:
[MIG] sale_delivery_split_date: Migration to 16.0

Commit:
[UPD] Update sale_automatic_workflow_job.pot

OCA-git-bot committed:
[UPD] README.rst

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 50.0% (3 of 6 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_order_warn_message
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_order_warn_message/it/

aktiv-heli-kantawala committed:
[MIG] sale_order_note_template: Migration to 16.0

aktiv-heli-kantawala committed:
[16.0-mig-sale_order_note_template][Fix] Fixed pre-commit

aktiv-heli-kantawala committed:
[MIG] sale_order_note_template: Resolve pre-commit

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_job
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_job/es/

manuelregidor committed:
[MIG]sale_order_warn_message: Migration to 15.0

Commit:
[UPD] Update sale_order_warn_message.pot

manuelregidor committed:
[MIG]sale_order_warn_message: Migration to 16.0

Commit:
[UPD] Update sale_order_warn_message.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_order_warn_message
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_order_warn_message/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_delivery_split_date
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_delivery_split_date/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_delivery_split_date
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_delivery_split_date/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (3 of 3 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_order_warn_message
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_order_warn_message/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_order_warn_message
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_order_warn_message/it/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_job
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_job/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_automatic_workflow_job
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_automatic_workflow_job/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (4 of 4 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_order_disable_user_autosubscribe
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_order_disable_user_autosubscribe/es/

MarinaAForgeFlow committed:
[ADD] sale_partner_address_restrict

siemenv committed:
[IMP] sale_delivery_split_date: pre-commit stuff

siemenv committed:
[MIG] sale_delivery_split_date: Migration to 17.0

sanderlienaerts committed:
[IMP] sale_order_lot_selection: black, isort, prettier

sanderlienaerts committed:
[MIG] sale_order_lot_selection: Migration to 17.0

LauraCForgeFlow committed:
[IMP] sale_delivery_split_date: use security lead time

Commit:
[UPD] Update sale_partner_address_restrict.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update sale_order_lot_selection.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-17.0/sale-workflow-17.0-sale_order_lot_selection
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-17-0/sale-workflow-17-0-sale_order_lot_selection/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: partner-contact-16.0/partner-contact-16.0-sale_partner_address_restrict
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-sale_partner_address_restrict/it/

lef-adhoc committed:
[IMP] sale_order_lot_selection: pre-commit execution

lef-adhoc committed:
[MIG] sale_order_lot_selection: Migration to 18.0

Commit:
[UPD] Update sale_delivery_split_date.pot

OCA-git-bot committed:
[BOT] post-merge updates

chaule97 committed:
[IMP] sale_automatic_workflow_job: pre-commit auto fixes

chaule97 committed:
[MIG] sale_automatic_workflow_job: Migration to 18.0

nguyenminhchien reviewed (changes requested):
We're migrating `sale_automatic_workflow` and its sub-modules `sale_automatic_workflow_*` because cosanum used them in version 14.0.

In cosanum 14.0, `sale_automatic_workflow`  includes the feature of `sale_automatic_workflow_stock`. So, we need to make sure cosanum 18.0 has that feature as well.

So, Please do 2 more tasks:
1. Migrate `sale_automatic_workflow_stock` to 18.0.
2. Introduce new module `sale_automatic_workflow_stock_job` which includes the feature related to job of stock.

xaviedoanhduy committed:
[IMP] sale_order_disable_user_autosubscribe: pre-commit auto fixes

xaviedoanhduy committed:
[MIG] sale_order_disable_user_autosubscribe: Migration to 18.0

chaule97 commented:
> We're migrating `sale_automatic_workflow` and its sub-modules `sale_automatic_workflow_*` because cosanum used them in version 14.0.
> 
> In cosanum 14.0, `sale_automatic_workflow` includes the feature of `sale_automatic_workflow_stock`. So, we need to make sure cosanum 18.0 has that feature as well.
> 
> So, Please do 2 more tasks:
> 
>     1. Migrate `sale_automatic_workflow_stock` to 18.0.
> 
>     2. Introduce new module `sale_automatic_workflow_stock_job` which includes the feature related to job of stock.

Hello @nguyenminhchien , I create new module [sale_automatic_workflow_stock_job](https://github.com/chaule97/sale-workflow/pull/5)

nguyenminhchien commented on sale_automatic_workflow_job/models/automatic_workflow_job.py:
```diff
@@ -0,0 +1,73 @@
+            "identity_key": identity_exact,
+        }
+
+    def _validate_sale_orders(self, domain_filter):
```
It breaks the flows of sending the order confirmation email. Look at the original method:
```
    @api.model
    def _validate_sale_orders(self, order_filter):
        sale_obj = self.env["sale.order"]
        sales = sale_obj.search(order_filter)
        _logger.debug("Sale Orders to validate: %s", sales.ids)
        for sale in sales:
            with savepoint(self.env.cr):
                self._do_validate_sale_order(
                    sale.with_company(sale.company_id), order_filter
                )
                if self.env.context.get("send_order_confirmation_mail"):
                    self._do_send_order_confirmation_mail(sale)
```
When `def_do_validate_sale_order()` does not change order's state to "sale" right away. 
So `def _do_send_order_confirmation_mail()` does not send email anymore

chaule97 replied:
I add `_do_send_order_confirmation_mail` in job and it will be runned after 2 minutes

JordiMForgeFlow committed:
[IMP] sale_partner_address_restrict: pre-commit fixes

JordiMForgeFlow committed:
[MIG] sale_partner_address_restrict: migration to V17

chaule97 committed:
[FIX] sale_automatic_workflow: use queue_job__no_delay

rousseldenis committed:
[IMP] sale_partner_address_restrict: pre-commit autofixes

rousseldenis committed:
[MIG] sale_partner_address_restrict: Migration to 18.0

rousseldenis committed:
[IMP] sale_partner_address_restrict: Be able to configure partner address restriction

rousseldenis committed:
[IMP] sale_partner_address_restrict: Use computed domain

ArnauCForgeFlow committed:
[IMP] sale_order_warn_message: pre-commit auto fixes

ArnauCForgeFlow committed:
[MIG] sale_order_warn_message: Migration to 18.0

ArnauCForgeFlow committed:
[IMP] sale_delivery_split_date: pre-commit auto fixes

ArnauCForgeFlow committed:
[MIG] sale_delivery_split_date: Migration to 18.0

IsabelAForgeFlow committed:
[MIG] sale_order_note_template: Migration to 16.0

twalter-c2c committed:
[ADD] sale_order_requested_delivery

twalter-c2c committed:
sale_order_requested_delivery: migration from 14.0

twalter-c2c committed:
sale_order_requested_delivery: pre-commit formatting

twalter-c2c committed:
sale_order_requested_delivery: fix field labels

twalter-c2c committed:
fixup! sale_order_requested_delivery: migration from 14.0

IsabelAForgeFlow committed:
[IMP] sale_order_note_template: pre-commit auto fixes

IsabelAForgeFlow committed:
[MIG] sale_order_note_template: Migration to 18.0

JordiMForgeFlow committed:
[IMP] sale_partner_address_restrict: mark dropship addresses to hide the main contact name

OCA-git-bot committed:
Merge PR #3409 into 18.0

Signed-off-by bodedra

Commit:
[UPD] Update sale_order_lot_selection.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3603 into 18.0

Signed-off-by dreispt

Commit:
[UPD] Update sale_delivery_split_date.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3602 into 18.0

Signed-off-by dreispt

Commit:
[UPD] Update sale_order_warn_message.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3453 into 18.0

Signed-off-by dreispt

Commit:
[UPD] Update sale_order_disable_user_autosubscribe.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3570 into 18.0

Signed-off-by rousseldenis

Commit:
[UPD] Update sale_partner_address_restrict.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3607 into 18.0

Signed-off-by rousseldenis

Commit:
[UPD] Update sale_order_note_template.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: sale-workflow-18.0/sale-workflow-18.0-sale_order_disable_user_autosubscribe
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-18-0/sale-workflow-18-0-sale_order_disable_user_autosubscribe/it/

OCA-git-bot committed:
Merge PR #3613 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update sale_order_requested_delivery.pot

OCA-git-bot committed:
[BOT] post-merge updates

chaule97 committed:
[FIX] sale_automatic_workflow_periodicity: use queue_job__no_delay
